Code/GraphMol/RGroupDecomposition/testRGroupDecomp.cpp

```diff
@@ -3196,6 +3196,258 @@ void testMultipleGroupsToUnlabelledCoreAtom() {
+  5  6  1  0  0  0  0
+  5  7  1  0  0  0  0
+  7  8  2  0  0  0  0
+  7  9  1  0  0  0  0
+  9 10  1  0  0  0  0
+ 10 11  1  1  0  0  0
+ 10 16  1  0  0  0  0
+ 10 12  1  0  0  0  0
+ 12 13  1  0  0  0  0
+ 13 14  1  0  0  0  0
+ 14 15  1  0  0  0  0
+ 15 16  1  0  0  0  0
+ 16 17  1  1  0  0  0
+ 16 18  1  0  0  0  0
+ 18 19  1  0  0  0  0
+ 14 20  1  0  0  0  0
+ 20 24  1  0  0  0  0
+ 20 21  2  0  0  0  0
+ 21 22  1  0  0  0  0
+ 22 23  2  0  0  0  0
+ 23 24  1  0  0  0  0
+ 23 25  1  0  0  0  0
+ 25 26  2  0  0  0  0
+ 25 27  1  0  0  0  0
+ 22 28  1  0  0  0  0
+ 28 33  2  0  0  0  0
+ 28 29  1  0  0  0  0
+ 29 30  2  0  0  0  0
+ 30 31  1  0  0  0  0
+ 31 32  2  0  0  0  0
+ 32 33  1  0  0  0  0
+ 31 34  1  0  0  0  0
+ 34 35  1  0  0  0  0
+ 35 36  1  0  0  0  0
+ 36 37  1  0  0  0  0
+ 37 42  1  0  0  0  0
+ 37 38  1  0  0  0  0
+ 38 39  1  0  0  0  0
+ 39 40  1  0  0  0  0
+ 40 41  1  0  0  0  0
+ 41 42  1  0  0  0  0
+ 40 43  1  0  0  0  0
+  4 44  1  0  0  0  0
+  3 45  1  0  0  0  0
+M  END
+)CTAB"_ctab;
+
+  RGroupDecompositionParameters params;
+  params.matchingStrategy = GreedyChunks;
+  params.allowMultipleRGroupsOnUnlabelled = false;
+  params.onlyMatchAtRGroups = true;
+  RGroupDecomposition decomp(*core, params);
+  auto result = decomp.add(*mol1);
+  TEST_ASSERT(result == 0);
+  result = decomp.add(*mol2);
+  TEST_ASSERT(result == 1);
+  decomp.process();
+  RGroupRows rows = decomp.getRGroupsAsRows();
+  TEST_ASSERT(rows.size() == 2) {
+    auto coreRgd = rows[0]["Core"];
+    auto match = std::find_if(
+        coreRgd->atoms().begin(), coreRgd->atoms().end(), [](Atom *a) {
+          return a->getAtomicNum() == 0 && a->getAtomMapNum() == 2;
+        });
+    TEST_ASSERT(match != coreRgd->atoms().end());
+    auto dummy = *match;
+    auto neighbor = *coreRgd->atomNeighbors(dummy).begin();
+    auto &conf = coreRgd->getConformer();
+    auto &dummyPoint = conf.getAtomPos(dummy->getIdx());
+    auto &neighborPoint = conf.getAtomPos(neighbor->getIdx());
+    auto length = (dummyPoint - neighborPoint).length();
+    TEST_ASSERT(abs(length - 1.0) < 0.005);
+    // R2 dummy should be directly above neighbor
+    TEST_ASSERT(abs(dummyPoint.x - neighborPoint.x) < 0.05);
+  }
+
+  {
+    auto coreRgd = rows[1]["Core"];
+    auto match = std::find_if(
+        coreRgd->atoms().begin(), coreRgd->atoms().end(), [](const Atom *a) {
+          return a->getAtomicNum() == 0 && a->getAtomMapNum() == 2;
+        });
+    TEST_ASSERT(match != coreRgd->atoms().end());
+    auto dummy = *match;
+    auto &conf = coreRgd->getConformer();
+    auto &dummyPoint = conf.getAtomPos(dummy->getIdx());
+    // R2 dummy should be over input chiral oxygen, which is first oxygen of degree 2 in input mol block
+    auto oIter = std::find_if(
+        mol2->atoms().begin(), mol2->atoms().end(), [](const Atom *a) {
+          return a->getAtomicNum() == 8 && a->getDegree() == 2;
+        });
+    TEST_ASSERT(oIter != mol2->atoms().end());
```

Comment:
Given that you're being this explicit about the conditions, it seems like you could just use the index of the O atom directly instead of doing this search.

Reply:
> Given that you're being this explicit about the conditions, it seems like you could just use the index of the O atom directly instead of doing this search.

I've made that change